### PR TITLE
re: Fix `clippy::clone-on_copy`

### PR DESCRIPTION
**Pull request by pmnoxx**
Fix `clippy::clone-on_copy` warnings.

Can be approved  by mm-near.

part of https://github.com/near/nearcore/pull/5882

### CODE REVIEW

File: integration-tests/src/tests/nearcore/track_shards.rs [hunk 1]
```diff
@@ -27,7 +27,7 @@ fn track_shards() {
         let last_block_hash: Arc<RwLock<Option<CryptoHash>>> = Arc::new(RwLock::new(None));
         WaitOrTimeoutActor::new(
             Box::new(move |_ctx| {
-                let bh = last_block_hash.read().unwrap().map(|h| h.clone());
+                let bh = last_block_hash.read().unwrap().map(|h| h);
```
Comment (matklad) [inline integration-tests/src/tests/nearcore/track_shards.rs:30]:
```suggestion
                let bh = last_block_hash.read().unwrap();
```

**Review by matklad — APPROVED** [review]

File: integration-tests/src/tests/nearcore/track_shards.rs (hunk 1 above)
Comment (pmnoxx) [reply integration-tests/src/tests/nearcore/track_shards.rs:30]: @matklad Reverted, that's not correct.
```
error[E0308]: mismatched types
  --> integration-tests/src/tests/nearcore/track_shards.rs:31:24
   |
31 |                 if let Some(block_hash) = bh {
   |                        ^^^^^^^^^^^^^^^^   -- this expression has type `std::sync::RwLockReadGuard<'_, Option<near_primitives::hash::CryptoHash>>`
   |                        |
   |                        expected struct `std::sync::RwLockReadGuard`, found enum `Option`
   |
   = note: expected struct `std::sync::RwLockReadGuard<'_, Option<near_primitives::hash::CryptoHash>, >`
                found enum `Option<_>`


```

Comment (pmnoxx) [reply integration-tests/src/tests/nearcore/track_shards.rs]: I replaced it with this
```
+                let bh = *last_block_hash.read().unwrap();
```

**Comment by pmnoxx** [comment]
@mm-near: /chain/ /nearcore/ /runtime/
```
@mm-near Can you take a look?

**Review by EgorKulikov — APPROVED** [review]

**Review by mm-near — APPROVED** [review]

**Pull request merged**